readthedocs/oauth/services/gitlab.py

```diff
@@ -179,9 +176,16 @@ def create_repository(self, fields, privacy=None, organization=None):
             else:
                 repo.clone_url = fields['http_url_to_repo']
 
-            repo.admin = not repo_is_public
-            if not repo.admin and 'owner' in fields:
-                repo.admin = self.is_owned_by(fields['owner']['id'])
+            # 0: No access
+            # 40: Maintainer
+            # 50: Owner
+            # https://docs.gitlab.com/ee/api/access_requests.html
+            # https://gitlab.com/help/user/permissions
+            project_access = fields.get('permissions', {}).get('project_access', {})
+            project_access_level = project_access.get('access_level', 0) if project_access else 0
```

Comment:
Looks like project access already defaults to an empty dict, this makes sense gitlab can set this to None. Same for group_access_level

```suggestion
            project_access_level = project_access.get('access_level', 0)
```

Reply:
I was bitten by this already, where `project_access` can be `None`. The `if` is to avoid bypass that case